### PR TITLE
Test cleanups

### PR DESCRIPTION
**Pull request by soc**
The first commit addresses the issue you found in https://github.com/dinfuehr/dora/pull/55#discussion_r312728526, the other three are unrelated cleanups I did while I was in that part of the code base.

### CODE REVIEW

File: tests/num-hash.dora [hunk 1]
```diff
@@ -0,0 +1,11 @@
+fun main() {
+    assert(true.hash() == 1);
+    assert(false.hash() == 0);
+    assert(1.toByte().hash() == 1);
+    assert(1.hash() == 1);
+    assert(1L.hash() == 1);
+    // thread 'main' panicked at 'fp-register accessed as gp-register.', src/baseline/expr.rs:52:18
```
Comment (dinfuehr) [inline tests/num-hash.dora]: This shouldn't panic anymore, right?

File: tests/array/array-contains.dora [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    let x = arrayFill[Bool](3, true);
+    assert(arrayContains[Bool](x, true));
+
+    let x = arrayFill[Int](3, 3);
+    assert(arrayContains[Int](x, 3));
+
+    let x = arrayFill[Long](3, 3L);
+    assert(arrayContains[Long](x, 3L));
+
+    // let x = arrayFill[Double](3, 0.0/0.0);
```
Comment (dinfuehr) [inline tests/array/array-contains.dora]: Does this still panic?

File: tests/string/string-hash.dora [hunk 3]
```diff
@@ -0,0 +1,9 @@
+fun main() {
+
+    // “target/debug/dora tests/array15…” terminated by signal SIGSEGV (Address boundary error)
```
Comment (dinfuehr) [inline tests/string/string-hash.dora]: Does this still crash?

File: tests/num-hash.dora (hunk 1 above)
Comment (soc) [reply tests/num-hash.dora]: No, will remove the comment. Thanks!

File: tests/array/array-contains.dora (hunk 2 above)
Comment (soc) [reply tests/array/array-contains.dora]: No, it shouldn't, will remove.

File: tests/string/string-hash.dora (hunk 3 above)
Comment (soc) [reply tests/string/string-hash.dora]: Yes, it does. :-/

**Comment by soc** [comment]
@dinfuehr rebased and updated.

**Pull request merged**